spark/v2.4/build.gradle

```diff
@@ -121,6 +121,7 @@ project(':iceberg-spark:iceberg-spark-runtime') {
       exclude group: 'org.xerial.snappy'
       exclude group: 'javax.xml.bind'
       exclude group: 'javax.annotation'
+      exclude group: 'io.netty'
```

Comment:
I feel we should still remove the unnecessary netty dependencies that spark already has on its runtime, and the correct thing to do here is:
```
exclude group: 'io.netty', module: 'netty-common'
```

@rdblue @pan3793 @ajantha-bhat what do you think?